### DISCUSSION TITLE
Document Architecture Decisions

### DISCUSSION THREAD

**Discussion by faselbaum** [deven-org/boiler#1091] [Polls]
Hey everybody 👋 

I would like to discuss if and how we should document architectural decisions going forward.

The current approach in `/docs/ARCHITECTURE.md` gives a rough overview but doesn't do much for the details and makes it hard to reason about the driving forces which led to these decisions.

I would really like to get the whole picture and why certain decisions where made to prevent us from accidentally losing key objectives along the way when we need to make structural changes in the future.

To tackle this issue i made a proposal in #1090
But before leaving your feedback there - i thought we should have a vote on this topic.

**Comment by thrbnhrtmnn** [comment]
Thanks @faselbaum for starting this discussion!
